java/maven/src/org/netbeans/modules/maven/modelcache/MavenProjectCache.java

```diff
@@ -350,6 +359,44 @@ private static boolean isLifecycleParticipatnIgnored(AbstractMavenLifecycleParti
         }
         return newproject;
     }
+
+    private static final String CONTEXT_PARTIAL_PROJECT = "org.netbeans.modules.maven.partialProject"; // NOI18N
+    private static final String CONTEXT_FALLBACK_PROJECT = "org.netbeans.modules.maven.fallbackProject"; // NOI18N
+    
+    /**
+     * Create a fallback project, but patch the incomplete project from the building result into it.
+     * The method will eventually start to return the partial project but still flagged as a fallback - see {@link #isFallbackproject(org.apache.maven.project.MavenProject)}.
+     * 
+     * @param result the maven execution / project building result.
+     * @param projectFile the project file.
+     * @return fallback project
+     * @throws AssertionError 
+     */
+    public static MavenProject getFallbackProject(MavenExecutionResult result, File projectFile) throws AssertionError {
+        MavenProject toReturn = getFallbackProject(projectFile);
+        if (result == null) {
+            return toReturn;
+        }
+        MavenProject partial = null;
+        
+        for (Throwable t : result.getExceptions()) {
```

Comment:
I wonder if lines 383 to 392 would fit better into MavenExecutionResult class ....

Reply:
I wonder how should I change library class.

Reply:
oh . sorry .. haven't noticed that - sure, you're right

Reply:
by the way, I whish Java had extention methods like C#.
It would be possible to write more elegant code.